server/intelligence-service/app/mentor/nodes/state_updates.py

```
@@ -0,0 +1,126 @@
+                if k != "messages" and k != "dev_progress"
+            ]
+        )
+        + "\n"
+    )
+    if state["closed"] or state["finish"] or state["goal_reflection"]:
+        # closed (finished) session state does not need to be updated
+        # when goal reflection is active, state is updated in the check_goal_reflection node
+        return
+
+    if state["goal_setting"]:
+        if state["last_thread"] == "":
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)